keras/backend/tensorflow_backend.py

```diff
@@ -570,6 +570,26 @@ def categorical_crossentropy(output, target, from_logits=False):
         return tf.nn.softmax_cross_entropy_with_logits(output, target)
 
 
+def categorical_crossentropy_one_hot(output, target, from_logits=False):
+    '''Note: tf.nn.sparse_softmax_cross_entropy_with_logits
+    expects logits, Keras expects prbabilities.
+    '''
+    if not from_logits:
+        output = tf.clip_by_value(output, tf.cast(_EPSILON, dtype=_FLOATX),
+                                  tf.cast(1.-_EPSILON, dtype=_FLOATX))
+        output = tf.log(output / (1 - output))
+
+    output_shape = output.get_shape()
+    res = tf.nn.sparse_softmax_cross_entropy_with_logits(
+        tf.reshape(output, [-1, int(output_shape[-1])]),
+        cast(flatten(target), 'int64'))
```

Comment:
this is working around the shape expectations of the tf function. This feels pretty ugly but it was the only way I was able to make this work with both timestep and single-target data. It was tested in the sense that I had toy models set up training on both types of data, and tested them with both backends and this was working.